build-system/tasks/e2e/describes-e2e.js

```diff
@@ -438,6 +471,10 @@ function describeEnv(factory) {
       });
 
       afterEach(async function () {
+        if (argv.coverage) {
+          await reportCoverage(env);
+        }
```

Comment:
`afterEach()` is called after every single test. Is the intent to report coverage once per test? If not, perhaps call `reportCoverage()` after all tests have completed, but before the server is stopped?

Reply:
I re-read the description and just noticed the bit about aggregating coverage. This is no longer a concern.

Reply:
Unresolving this thread because making an XHR after each test is a concern of mine as well.

Reply:
@estherkim Addressed in comment below, PTAL

Reply:
Even though this work is experimental as you say, I think this part is worth iterating over. How about writing to a local file?

Reply:
What are you suggesting we write to a local file?

If you're referring to the coverage data itself, the issue isn't just storing it, it's actually merging coverage objects across multiple runs. This functionality is built into the `istanbul-middleware` package that handles the coverage UI. I could presumably reverse-engineer that logic (there's an attempt at that elsewhere, though it's not maintained), or there may be a way to get into the plugin logic itself. I would prefer not to embed that attempt & complexity in this PR, instead committing this to iterate on as a base point.

Reply:
@estherkim @rsimha I'd really like to resolve this this week if possible so I can make progress in adjacent areas without the code changing underneath

Reply:
Totally fair, maybe @rsimha can weigh in since you and I seem to be at an impasse on what's acceptable as experimental :)

Reply:
To add color, e2e tests each make multiple HTTP requests _to and from the browser_, while this adds one request _that takes place within Node, not the browser_. There is no apparent impact on runtime with vs. without this coverage reporting.

Reply:
Good point, okay then. Approving this to unblock.

Reply:
Thanks :+1:

Reply:
Just caught up with the comments. Thanks for iterating on this @rcebulko and @estherkim!

Here are the three highlights that stood out to me:

> 1. _"I'd prefer to get this in with the understanding that it's experimental and will never be enabled in builds while the XHR-per-test is still in place."_
> 2. "_In parallel, I plan to modify init_tests to not clear the coverage variable globally, so that the coverage info can be reported one single time at the end of the full test suite._"
> 3. "_There is no apparent impact on runtime with vs. without this coverage reporting._"

Based on these three facts, I'm comfortable with the current code being checked in.